diff -Naur a/prom/include/prom_collector.h b/prom/include/prom_collector.h
--- a/prom/include/prom_collector.h	2020-12-08 02:38:13.000000000 -0500
+++ b/prom/include/prom_collector.h	2024-04-22 10:44:18.385709848 -0400
@@ -88,6 +88,14 @@
 int prom_collector_add_metric(prom_collector_t *self, prom_metric_t *metric);
 
 /**
+ * @brief Remove a metric from a collector
+ * @param self The target prom_collector_t*
+ * @param metric the prom_metric_t* to remove from the prom_collector_t* passed as self.
+ * @return A non-zero integer value upon failure.
+ */
+int prom_collector_remove_metric(prom_collector_t *self, prom_metric_t *metric);
+
+/**
  * @brief The collect function is responsible for doing any work involving a set of metrics and then returning them
  *        for metric exposition.
  * @param self The target prom_collector_t*
diff -Naur a/prom/src/prom_collector.c b/prom/src/prom_collector.c
--- a/prom/src/prom_collector.c	2020-12-08 02:38:13.000000000 -0500
+++ b/prom/src/prom_collector.c	2024-04-22 10:58:04.280525409 -0400
@@ -115,6 +115,12 @@
   return prom_map_set(self->metrics, metric->name, metric);
 }
 
+int prom_collector_remove_metric(prom_collector_t *self, prom_metric_t *metric) {
+  PROM_ASSERT(self != NULL);
+  if (self == NULL) return 1;
+  return prom_map_delete(self->metrics, metric->name);
+}
+
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // Process Collector
 
