diff -Naur Pktgen-DPDK-pktgen-23.03.0-pristine/app/pktgen-port-cfg.c Pktgen-DPDK-pktgen-23.03.0/app/pktgen-port-cfg.c
--- Pktgen-DPDK-pktgen-23.03.0-pristine/app/pktgen-port-cfg.c	2026-03-10 13:43:29.373015422 +0000
+++ Pktgen-DPDK-pktgen-23.03.0/app/pktgen-port-cfg.c	2026-03-10 14:05:31.581155195 +0000
@@ -248,7 +248,7 @@
         info->seq_pkt = rte_zmalloc_socket(buff, (sizeof(pkt_seq_t) * NUM_TOTAL_PKTS),
                                            RTE_CACHE_LINE_SIZE, rte_socket_id());
         if (info->seq_pkt == NULL)
-            pktgen_log_panic("Unable to allocate %d pkt_seq_t headers", NUM_TOTAL_PKTS);
+            pktgen_log_panic("Unable to allocate %ld pkt_seq_t headers", NUM_TOTAL_PKTS);
 
         for (int i = 0; i < NUM_TOTAL_PKTS; i++) {
             info->seq_pkt[i].seq_enabled = 1;
